pkg/mqtrigger/kafka/main.go

```diff
@@ -0,0 +1,480 @@
+)
+
+var SHA256 scram.HashGeneratorFcn = func() hash.Hash { return sha256.New() }
+var SHA512 scram.HashGeneratorFcn = func() hash.Hash { return sha512.New() }
+
+type XDGSCRAMClient struct {
+	*scram.Client
+	*scram.ClientConversation
+	scram.HashGeneratorFcn
+}
+
+func (x *XDGSCRAMClient) Begin(userName, password, authzID string) (err error) {
+	x.Client, err = x.HashGeneratorFcn.NewClient(userName, password, authzID)
+	if err != nil {
+		return err
+	}
+	x.ClientConversation = x.Client.NewConversation()
+	return nil
+}
+
+func (x *XDGSCRAMClient) Step(challenge string) (response string, err error) {
+	response, err = x.ClientConversation.Step(challenge)
+	return
+}
+
+func (x *XDGSCRAMClient) Done() bool {
+	return x.ClientConversation.Done()
+}
+
+func parseKafkaMetadata(logger *zap.Logger) (kafkaMetadata, error) {
+	meta := kafkaMetadata{}
+
+	// brokerList marked as deprecated, bootstrapServers is the new one to use
+	if os.Getenv("BROKER_LIST") != "" && os.Getenv("BOOTSTRAP_SERVERS") != "" {
+		return meta, errors.New("cannot specify both bootstrapServers and brokerList (deprecated)")
+	}
+	if os.Getenv("BROKER_LIST") == "" && os.Getenv("BOOTSTRAP_SERVERS") == "" {
+		return meta, errors.New("no bootstrapServers or brokerList (deprecated) given")
+	}
+	if os.Getenv("BOOTSTRAP_SERVERS") != "" {
+		meta.bootstrapServers = strings.Split(os.Getenv("BOOTSTRAP_SERVERS"), ",")
+	}
+	if os.Getenv("BROKER_LIST") != "" {
+		logger.Info("WARNING: usage of brokerList is deprecated. use bootstrapServers instead.")
+		meta.bootstrapServers = strings.Split(os.Getenv("BROKER_LIST"), ",")
+	}
+
+	meta.authMode = kafkaAuthModeForNone
+	mode := kafkaAuthMode(strings.TrimSpace((os.Getenv("AUTH_MODE"))))
+	if mode == "" {
+		mode = kafkaAuthModeForNone
+	}
+
+	if mode != kafkaAuthModeForNone && mode != kafkaAuthModeForSaslPlaintext && mode != kafkaAuthModeForSaslSSL && mode != kafkaAuthModeForSaslSSLPlain && mode != kafkaAuthModeForSaslScramSha256 && mode != kafkaAuthModeForSaslScramSha512 {
+		return meta, fmt.Errorf("err auth mode %s given", mode)
+	}
+
+	meta.authMode = mode
+
+	if meta.authMode != kafkaAuthModeForNone && meta.authMode != kafkaAuthModeForSaslSSL {
+		if os.Getenv("USERNAME") == "" {
+			return meta, errors.New("no username given")
+		}
+		meta.username = strings.TrimSpace(os.Getenv("USERNAME"))
+
+		if os.Getenv("PASSWORD") == "" {
+			return meta, errors.New("no password given")
+		}
+		meta.password = strings.TrimSpace(os.Getenv("PASSWORD"))
+	}
+
+	if meta.authMode == kafkaAuthModeForSaslSSL {
+		if os.Getenv("CA") == "" {
+			return meta, errors.New("no ca given")
+		}
+		meta.ca = os.Getenv("CA")
+
+		if os.Getenv("CERT") == "" {
+			return meta, errors.New("no cert given")
+		}
+		meta.cert = os.Getenv("CERT")
+
+		if os.Getenv("KEY") == "" {
+			return meta, errors.New("no key given")
+		}
+		meta.key = os.Getenv("KEY")
+	}
+
+	return meta, nil
+}
+
+func parseFissionMetadata() (fissionMetadata, error) {
```

Comment:
Similar to comment about the fissionMetadata struct - this method can potentially go to a common place outside of Kafka

Reply:
Done